gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/index/GfxdIndexManager.java

```diff
@@ -438,8 +416,15 @@ public void onEvent(final LocalRegion owner, final EntryEventImpl event,
               owner.getFullPath());
         }
 
+        boolean isRecovered = false;
+        if (!owner.isInitialized()) {
+          DiskRegion dr = owner.getDiskRegion();
+          if (dr != null) {
+            isRecovered = dr.testIsRecovered(entry, false);
+          }
+        }
         boolean throwEEE = false;
-        if (!posDup && owner.isInitialized()) {
+        if (!posDup && (owner.isInitialized() || (!owner.isInitialized() && !isRecovered))) {
```

Comment:
@suranjan
a) Ok, so I guess better option will be to mark somewhere (in RegionEntry or EntryEventImpl?) so that change does not get applied. Alternatively higher layer in GemFire can catch the EEE and ignores it when node is not initialized but still skip applying the change. Reason below.

b) Ok, still leaves one question. This whole check to skip throwing EEE is when there is a genuine case where same update has come in through both GII as well as directly. With this change such cases can now throw spurious EEE. Hence the comments in point a).